sympy/core/mod.py

```diff
@@ -63,7 +63,7 @@ def number_eval(p, q):
                         return S.One
 
             if hasattr(p, '_eval_Mod'):
-                rv = getattr(p, '_eval_Mod')(q)
+                rv = p._eval_Mod(q)
```

Comment:
I seem to remember someone clearing out things like this based on some previous automated tool. I guess new ones can always creep back in.

Reply:
It would be better to change this to
```python
_eval_Mod = getattr(p, '_eval_Mod', None)
if _eval_Mod is not None:
    rv = p._eval_Mod(q)
```
That only needs a single attribute lookup.